pkg/framework/status.go

```diff
@@ -0,0 +1,80 @@
+package framework
+
+import (
+	"github.com/orelmisan/k8s-checkup-framework/pkg/checkup"
+	"strconv"
+	"strings"
+	"time"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+const (
+	statusSucceededKey        = "status.succeeded"
+	statusFailureReasonKey    = "status.failureReason"
+	statusStartTimestamp      = "status.startTimestamp"
+	statusCompletionTimestamp = "status.completionTimestamp"
+	statusResultNameKeyPrefix = "status.result."
+)
+
+type Status struct {
+	succeeded           string
+	failureReason       string
+	startTimestamp      string
+	completionTimestamp string
+	results             map[string]string
+}
+
+func NewStatus() *Status {
+	return &Status{
+		succeeded:     "",
+		failureReason: "",
+	}
+}
+
+func (s *Status) SetSucceeded(v bool) {
+	s.succeeded = strconv.FormatBool(v)
+}
+
+func (s *Status) SetFailureReason(v string) {
+	s.failureReason = v
+}
```

Comment:
I use the the matching getter, it will be confusing to use getter/setter on one field and a plain field on another.

Reply:
I am not sure I follow, where is the matching getter?
I think it would make sense to you if `Status` properties were not plain strings as mentioned at https://github.com/orelmisan/k8s-checkup-framework/pull/9/commits/29cd2698211c1e2c45f5016f93f2b1d4bc59cec0#r822761465
For example you would not need for the conversion at `SetSucceeded`, it could have been exported instead.
Having said that, if you cant find the time to address it now we can do this later.